### PR TITLE
[MIGRATION] Purge FAILED_JOIN_QUEUE & add queues to query

### PR DESCRIPTION
**Pull request by njerschow**

### CODE REVIEW

**Review by nseguias — APPROVED** [review]
lgtm, missing changelog

File: smart-contracts/contracts/lp-strategy/src/contract.rs [hunk 1]
```diff
@@ -501,6 +501,11 @@ pub fn migrate(deps: DepsMut, _env: Env, msg: MigrateMsg) -> Result<Response, Co
     // add a new fresh unlocked ibc lock
     IBC_LOCK.save(deps.storage, &Lock::new())?;
 
+    // goodbye join queue
+    while !FAILED_JOIN_QUEUE.is_empty(deps.storage)? {
+        FAILED_JOIN_QUEUE.pop_front(deps.storage)?;
```
Comment (LaurensKubat) [inline smart-contracts/contracts/lp-strategy/src/contract.rs:506]: We shoudl re-add these to trapped errors as a failed join pool so we don't lose them

**Review by LaurensKubat — CHANGES_REQUESTED** [review]

**Review by LaurensKubat — DISMISSED** [review]
lgtm

**Review by LaurensKubat — APPROVED** [review]
lgtm

**Review by nseguias — APPROVED** [review]
lgtm

**Pull request merged**